app/src/main/java/net/bonysoft/magicmirror/MainActivity.java

```diff
@@ -137,4 +155,40 @@ private DecimalFormat getTemperatureFormat() {
         temperatureFormat.setRoundingMode(RoundingMode.HALF_DOWN);
         return temperatureFormat;
     }
+
+    private final TwitterModule.TwitterListener tweetListener = new TwitterModule.TwitterListener() {
+
+        @Override
+        public void onNextTweet(final Status tweet) {
+            runOnUiThread(new Runnable() {
```

Comment:
Mmm honestly I don't think it's responsibility of the caller to do this.
The only visibility of the caller is the interface, then the implementer should take care of this, through a decorator maybe?
I'd suggest to keep this for now, as we would need to clean this activity anyway, using a proper presenter